Fix check health

relardev opened the pull request:
On nvim v0.9.5 when trying to run `:checkhealth` I kept getting lua errors from kickstart

1. the `vim.health....` functions should all start with `report_`, help results for `vim.health`: 
![image](https://github.com/nvim-lua/kickstart.nvim/assets/154016947/d080d324-b7a9-4225-817c-c58ee490a08e)

1. `vim.version.cmp` expects 2 strings but was getting 2 tables:
![image](https://github.com/nvim-lua/kickstart.nvim/assets/154016947/0b367274-9825-44ad-a8a7-f1bc1957e363)

Not sure if this are correct fixes, this is what I had to do to make `:checkhealth` pass.

dam9000 commented:
@relardev what kind of lua errors are you getting? Can you post a log or screenshot of the errors?
I also have nvim 0.9.5 and I'm not seeing any errors:
```
kickstart: require("kickstart.health").check()

kickstart.nvim ~
- NOTE: Not every warning is a 'must-fix' in `:checkhealth`
  
  Fix only warnings for plugins and languages you intend to use.
  Mason will give warnings for languages that are not installed.
  You do not need to install, unless you want to use those languages!
- System Information: {
  machine = "x86_64",
  release = "6.7.6-200.fc39.x86_64",
  sysname = "Linux",
  version = "#1 SMP PREEMPT_DYNAMIC Fri Feb 23 18:27:29 UTC 2024"
  }
- OK Neovim version is: '0.9.5'
- OK Found executable: 'git'
- OK Found executable: 'make'
- OK Found executable: 'unzip'
- OK Found executable: 'rg'
```

dam9000 commented:
Also in health.lua there is:
```
function M.start(msg)
  M.report_start(msg)
end
function M.info(msg)
  M.report_info(msg)
end
function M.ok(msg)
  M.report_ok(msg)
end
function M.warn(msg, ...)
  M.report_warn(msg, ...)
end
function M.error(msg, ...)
  M.report_error(msg, ...)
end
```
So the code as is should work as is, although, I guess the shorthand functions were added later and if we would want to report an error properly for older versions, the code should indeed be adapted to the old APIs. Buf this further points to your version being older than 0.9.5.

relardev commented:
```
> nvim --version

NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1692716794

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/__w/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

Run :checkhealth for more info
```

dam9000 commented:
@relardev did you build the nvim manually or is it installed from a package? Is it possible you have both - a system installed nvim and a local build and the runtimes clash? Which OS do you use?

relardev commented:
I updated the nvim today from 0.9.1 by moving new binary into old one(previously I updated it the same way from who knows what version), but didn't touch the share files, didn't know they need to be updated too. My expectation was that the neovim binary has all that it needs. Sorry for fake pr, at least i learned something
